src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -60,7 +60,7 @@ void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::ResetData(
   numFunctions = responses.n_cols;
   this->predictors = std::move(predictors);
   this->responses = std::move(responses);
-  this->deterministic = true;
+  this->deterministic = false;
```

Comment:
ResetData is used in training mode and hence deterministic should be set to false so that all layers are in training mode.

Reply:
Nice catch!